### PR TITLE
fix(web): context-only rule effects, set(&layer) from physical keystrokes 🖇️ 🍒

### PR DESCRIPTION
**Pull request by jahorton**
A combo-🍒 for both #6901 and #6902 - so, the full 🖇️ set.

There is one notable change from #6902, based on this comment: https://github.com/keymanapp/keyman/pull/6902#discussion_r916426897

Just in case we'd rather split this into two separate 🍒s, commit https://github.com/keymanapp/keyman/pull/6949/commits/9ffca15f643ead60efc810d454d2335c343d14ea is the final one from the first PR.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/source/kmwbase.ts [hunk 1]
```diff
@@ -181,15 +181,26 @@ namespace com.keyman {
...
+    /**
+     * Reset context when entering or exiting the active element.
+     * Will also trigger OSK shift state / layer reset.
+     **/
+    pageFocusHandler = () => {
+      if(!this.uiManager.isActivating && this.osk?.vkbd && this.core.contextDevice.touchable) {
+        this.core.keyboardProcessor._UpdateVKShift(null);
+      }
+      return false;
+    }
```
Comment (jahorton) [inline web/source/kmwbase.ts:193]: Note lines 189 & 190 - they differ slightly from the 16.0 version in order to keep this change a little closer to the original stable-15.0 code.

Reference https://github.com/keymanapp/keyman/pull/6902#discussion_r916426897 and its chain for the reasoning.

File: web/source/kmwbase.ts [hunk 2]
```diff
@@ -181,15 +181,26 @@ namespace com.keyman {
...
+    /**
+     * Reset context when entering or exiting the active element.
+     * Will also trigger OSK shift state / layer reset.
+     **/
+    pageFocusHandler = () => {
+      if(!this.uiManager.isActivating && this.osk?.vkbd && this.util.device.touchable) {
+        this.core.keyboardProcessor._UpdateVKShift(null);
+      }
+      return false;
+    }
```
Comment (jahorton) [inline web/source/kmwbase.ts:193]: Note lines 189 & 190 - they differ slightly from the 16.0 version in order to keep this change a little closer to the original stable-15.0 code.

Reference https://github.com/keymanapp/keyman/pull/6902#discussion_r916426897 and its chain for the reasoning.

**Review by mcdurdin — APPROVED** [review]
LGTM but can we at least run a couple of sanity user tests on this -- I don't see any reason it would fail, but it wouldn't hurt to be sure.

**Comment by jahorton** [comment]
Ack!  Thought I was merging a different 🍒!

Still need to run a couple of sanity user tests on it as suggested; major apologies there.

**Comment by jahorton** [comment]
Test issue @ #6953.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.267](https://keyman.com/downloads/releases/stable/15.0.267)

**Pull request merged**